diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/BooleanExpressionsRule.kt

```diff
@@ -0,0 +1,208 @@
+package org.cqfn.diktat.ruleset.rules.chapter3
+
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.COMPLEX_BOOLEAN_EXPRESSION
+import org.cqfn.diktat.ruleset.rules.DiktatRule
+import org.cqfn.diktat.ruleset.utils.KotlinParser
+import org.cqfn.diktat.ruleset.utils.findAllNodesWithCondition
+import org.cqfn.diktat.ruleset.utils.findLeafWithSpecificType
+
+import com.bpodgursky.jbool_expressions.Expression
+import com.bpodgursky.jbool_expressions.parsers.ExprParser
+import com.bpodgursky.jbool_expressions.rules.RuleSet
+import com.pinterest.ktlint.core.ast.ElementType
+import com.pinterest.ktlint.core.ast.ElementType.BINARY_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.OPERATION_REFERENCE
+import com.pinterest.ktlint.core.ast.ElementType.PARENTHESIZED
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+import java.lang.RuntimeException
+
+/**
+ * Rule that warns if the boolean expression can be simplified.
+ */
+class BooleanExpressionsRule(configRules: List<RulesConfig>) : DiktatRule(
+    "boolean-expressions-rule",
+    configRules,
+    listOf(COMPLEX_BOOLEAN_EXPRESSION)) {
+    override fun logic(node: ASTNode) {
+        if (node.elementType == ElementType.CONDITION) {
+            checkBooleanExpression(node)
+        }
+    }
+
+    @Suppress("TooGenericExceptionCaught")
+    private fun checkBooleanExpression(node: ASTNode) {
+        // This map is used to assign a variable name for every elementary boolean expression.
+        val mapOfExpressionToChar: HashMap<String, Char> = HashMap()
+        val correctedExpression = makeCorrectExpressionString(node, mapOfExpressionToChar)
+        // If there are method calls in conditions
+        val expr: Expression<String> = try {
+            ExprParser.parse(correctedExpression)
+        } catch (exc: RuntimeException) {
```

Comment:
What do you mean? Can you provide an example?

Reply:
What type of exception does this library throw, when we provide an uparseable expression? Catching `RuntimeException` seems too generice, since we can silently ignore exceptions not related to parsing. If they *really* throw just RTE, then ok.

Reply:
Yes, it throws:
``` kotlin
throw new RuntimeException("Unrecognized! " + tree.getType() + " " + tree.getText());
```

Reply:
ok... maybe let's try to check exception's message then?